coalib/bearlib/abstractions/Linter.py

```diff
@@ -676,18 +676,50 @@ def run(self, filename=None, file=None, **kwargs):
                 self.debug("Running '{}'".format(
                     ' '.join(str(arg) for arg in arguments)))
 
-                output = run_shell_command(
+                result = run_shell_command(
                     arguments,
                     stdin=''.join(file) if options['use_stdin'] else None,
                     cwd=self.get_config_dir())
 
-                output = tuple(compress(
-                    output,
-                    (options['use_stdout'], options['use_stderr'])))
+                stdout, stderr = result
+
+                output = []
+
+                if options['use_stdout']:
+                    output.append(stdout)
+                elif stdout:
+                    logging.warning(
+                        '{}: Discarded stdout: {}'.format(
+                            self.__class__.__name__, stdout))
+                    stdout = ''
+
+                if options['use_stderr']:
+                    output.append(stderr)
+                elif stderr:
+                    logging.warning(
+                        '{}: Discarded stderr: {}'.format(
+                            self.__class__.__name__, stderr))
+                    stderr = ''
+
+                if result.code:
+                    logging.warning(
+                        '{}: Exit code {}'.format(
+                            self.__class__.__name__, result.code))
+
                 if options['strip_ansi']:
                     output = tuple(map(strip_ansi, output))
+                else:
+                    output = tuple(output)
+
                 if len(output) == 1:
                     output = output[0]
+
+                if not any([stdout, stderr]):
```

Comment:
Put it between `if result.code` and `if options['strip_ansi']`?
Since `map(strip_ansi, output))` work becomes unnecessary if condition is true here.

Reply:
Good call.

I also moved things around a bit.